tests/test_debugging.py

```diff
@@ -376,3 +379,166 @@ def circuit():
             return qml.expval(qml.PauliZ(0))
 
         qml.snapshots(circuit)()
+
+
+# pylint: disable=protected-access
+class TestPLDB:
+    """Test the interactive debugging integration"""
+
+    def test_pldb_init(self):
+        """Test that PLDB initializes correctly"""
+        debugger = PLDB()
+        assert debugger.prompt == "[pldb]: "
+        assert getattr(debugger, "_PLDB__active_dev") is None
+
+    def test_valid_context_outside_qnode(self):
+        """Test that valid_context raises an error when breakpoint
+        is called outside of a qnode execution."""
+
+        with pytest.raises(
+            RuntimeError, match="Can't call breakpoint outside of a qnode execution"
+        ):
+            with qml.queuing.AnnotatedQueue() as _:
+                qml.X(0)
+                qml.breakpoint()
+                qml.Hadamard(0)
+
+        def my_qfunc():
+            qml.X(0)
+            qml.breakpoint()
+            qml.Hadamard(0)
+            return qml.expval(qml.Z(0))
+
+        with pytest.raises(
+            RuntimeError, match="Can't call breakpoint outside of a qnode execution"
+        ):
+            _ = my_qfunc()
+
+    def test_valid_context_not_compatible_device(self):
+        """Test that valid_context raises an error when breakpoint
+        is called with an un-supported device."""
+        dev = qml.device("default.mixed", wires=2)
+
+        @qml.qnode(dev)
+        def my_circ():
+            qml.X(0)
+            qml.breakpoint()
+            qml.Hadamard(0)
+            return qml.expva(qml.Z(0))
+
+        with pytest.raises(TypeError, match="Breakpoints not supported on this device"):
+            _ = my_circ()
+
+        PLDB.reset_active_dev()
+
+    def test_add_device(self):
+        """Test that we can add a device to the global active device list."""
+        assert getattr(PLDB, "_PLDB__active_dev") is None
+
+        dev1, dev2, dev3 = (
+            qml.device("default.qubit", wires=3),
+            qml.device("default.qubit"),
+            qml.device("lightning.qubit", wires=1),
+        )
+
+        PLDB.add_device(dev1)
+        assert getattr(PLDB, "_PLDB__active_dev") == dev1
+
+        PLDB.add_device(dev2)  # overwrites dev1
+        PLDB.add_device(dev3)  # overwrites dev2
+
+        assert getattr(PLDB, "_PLDB__active_dev") == dev3
+
+        PLDB.reset_active_dev()  # clean up the debugger active devices
+
+    dev_names = (
+        "default.qubit",
+        "lightning.qubit",
+    )
+
+    @pytest.mark.parametrize("device_name", dev_names)
+    def test_get_active_device(self, device_name):
+        """Test that we can accses the active device."""
+        dev = qml.device(device_name, wires=2)
+        PLDB.add_device(dev)
+
+        debugger_dev = PLDB.get_active_device()
+        assert debugger_dev is dev
+
+        PLDB.reset_active_dev()
```

Comment:
# Question
Why are we calling this here? If this is done as part of a test clean-up, we should instead make this a `fixture` with `autouse=True` which calls this function after a `yield`.

Reply:
It is just test clean-up. This was written before we created the context manager, I can just use that for these tests

Reply:
Added